mailbox/api/src/main/scala/org/apache/james/mailbox/acl/ACLDiff.scala

```diff
@@ -0,0 +1,58 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.mailbox.acl
```

Comment:
Are we expecting people reading and using such an important core API to understand scala?

This don't fit my definition of "component"

Reply:
You are right.

However, ACLDiff is not my definition of API.

Could we move toward ACL implementation being a component?

Reply:
It's exposed in the API via MailboxACLUpdated (MailboxListener class), which a user can rely on for impementing custom listeners. Hence it don't fully fit the definition I have of ACL implementation. 

Though I agree the degree of exposure is low it still, technically is the mailbox API.

Hence my real concerns.